src/gui/editors/SongEditor.cpp

```diff
@@ -64,31 +62,21 @@ namespace lmms::gui
 
 namespace
 {
-constexpr int ZOOM_MIN = 15;
-constexpr int ZOOM_MAX = 1600;
-constexpr int ZOOM_INI = 100;
-}
 
-int scaleToLinearZoom(int val)
-{
-	// linear: val should be in min - max range.
-	float linVal = linearToLogScale(ZOOM_MIN, ZOOM_MAX, val);
-	return std::clamp(static_cast<int>(std::round(linVal)), ZOOM_MIN, ZOOM_MAX);
-}
+constexpr int MIN_PIXELS_PER_BAR = 2;
+constexpr int MAX_PIXELS_PER_BAR = 400;
+constexpr int ZOOM_STEPS = 200;
+
+constexpr std::array SNAP_SIZES{8.f, 4.f, 2.f, 1.f, 1/2.f, 1/4.f, 1/8.f, 1/16.f};
```

Comment:
I opted to define snap sizes manually instead of calculating them on three different places. This opens the possibility of snap sizes like 1/3 etc